pkgs/development/python-modules/ruff-lsp/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib
+, stdenv
+, pythonOlder
+, buildPythonPackage
+, fetchPypi
+, ruff
+, pygls
+, lsprotocol
+, hatchling
+, typing-extensions
+, unittestCheckHook
+, python-lsp-jsonrpc
+}:
+
+buildPythonPackage rec {
+  pname = "ruff-lsp";
+  version = "0.0.24";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "ruff_lsp";
+    sha256 = "sha256-1he/GYk8O9LqPXH3mu7eGWuRygiDG1OnJ+JNT2Pynzo=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+
+  nativeCheckInputs = [
+    unittestCheckHook
+    python-lsp-jsonrpc
+    ruff
+  ];
```

Comment:
And this under `doCheck`.

Reply:
Are these requests personal preferences or documented somewhere as best practices?